File: frontend/full-stack-project/src/app/contract/Ticket.ts
export interface Ticket {
  customerName?: any;
  id: number;
  title: string;
  description: string;
  priority: 'LOW' | 'MEDIUM' | 'HIGH';
  status: 'OPEN' | 'IN_PROGRESS' | 'PENDING' | 'RESOLVED';
  userId: number;
  assignedToId: number | null;
  createdAt: string;
  updatedAt: string;
}

File: backend/src/main/java/com/dollop/app/model/Ticket.java
package com.dollop.app.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Ticket {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	private String title;
	private String description;
	
	@Enumerated(EnumType.STRING)
	private Priority priority;
	
	@Enumerated(EnumType.STRING)
	private Status status;
	@ManyToOne
    @JoinColumn(name = "assigned_to_id")
    private User assignedTo;
	
	@ManyToOne
	private User user;
	private LocalDateTime createdAt;
	private LocalDateTime updatedAt;
	
	
}
